demux/demux.c

```diff
@@ -1084,14 +1085,20 @@ static void demux_shutdown(struct demux_internal *in)
     demuxer->priv = NULL;
     in->d_thread->priv = NULL;
 
-    demux_flush(demuxer);
-    assert(in->total_bytes == 0);
+    // Intentionally leak demuxer cache to avoid long and unnecessary
+    // deallocation of possibly hundreds of thousands of cached packets when
+    // exiting the player. Note that leak checkers may complain about this, so
+    // they have to be configured appropriately.
+    if (!in->shutdown_fast) {
+        demux_flush(demuxer);
+        assert(in->total_bytes == 0);
 
-    in->current_range = NULL;
-    free_empty_cached_ranges(in);
+        in->current_range = NULL;
+        free_empty_cached_ranges(in);
 
-    talloc_free(in->cache);
-    in->cache = NULL;
+        talloc_free(in->cache);
+        in->cache = NULL;
+    }
```

Comment:
Nit: I'd still add a log message here on the `else` branch.

But otherwise, I think the current patch is probably in a good state.

Reply:
I really don't want to advertise this implementation detail to uneducated users.

Reply:
By "log" I meant something like a debug log (e.g `MP_DBG` or `MP_TRACE`) which aren't enabled by default. But drop it if you don't think it's worth it.